bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -509,7 +533,31 @@ func (r *KubeadmConfigReconciler) joinWorker(ctx context.Context, scope *Scope)
 			UseExperimentalRetry: scope.Config.Spec.UseExperimentalRetryJoin,
 		},
 		JoinConfiguration: joinData,
-	})
+	}
+
+	var cloudJoinData []byte
+
+	switch scope.Config.Spec.Format {
+	case bootstrapv1.Ignition:
+		ign, warnings, err := ignition.NewNode(&ignition.NodeInput{
```

Comment:
I think that accepting a KubeadmConfig object and then failing at rendering later is not ideal as a user experience.
What about catching those problems at validation time?

Reply:
I think discussion in https://github.com/kubernetes-sigs/cluster-api/pull/4172#discussion_r583750268 covers this.